### PR TITLE
Graphql directive

### PR DESCRIPTION
**Pull request by AleF83**
This PR adds new directive that allows to `gateway` to resolve data from remote graphql server like it already does from remote rest server with `@http` directive

### CODE REVIEW

File: services/graphql-server/directives/common/gql.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	upstream, ok := s.Upstream(url.Host)
+	if ok {
+		upstream.ApplyUpstream(rp.Context, &request.Header)
+	}
+	return request, nil
+}
+
+func sendRequest(request *gqlclient.Request, client *gqlclient.Client, gqlParams gqlParams) (interface{}, error) {
+	ctx := context.Background()
```
Comment (AvivRubys) [inline services/graphql-server/directives/common/gql.go]: Use context from ResolveParams

File: services/graphql-server/directives/common/gql.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	var respData interface{}
+	if err := client.Run(ctx, request, &respData); err != nil {
+		logrus.WithError(err).Panic("error while graphql request")
```
Comment (AvivRubys) [inline services/graphql-server/directives/common/gql.go]: No reason to panic when we can just return the error or wrap it

File: services/graphql-server/directives/common/gql.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	result := respData.(map[string]interface{})[gqlParams.queryName]
```
Comment (AvivRubys) [inline services/graphql-server/directives/common/gql.go]: Possible panic here, check type conversion with second argument

File: services/graphql-server/utils/ast2sdl.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/graphql-go/graphql"
+	"github.com/graphql-go/graphql/language/ast"
+)
+
+// ResolveParamsToSDLQuery - generates SDL query from graphql.ResolveParams object
+func ResolveParamsToSDLQuery(queryName string, rp graphql.ResolveParams) string {
```
Comment (AvivRubys) [inline services/graphql-server/utils/ast2sdl.go]: This could use a unit test or two

File: services/registry/Dockerfile [hunk 5]
```diff
@@ -2,6 +2,11 @@ FROM node:10-alpine as base
 WORKDIR /service
 
 FROM base as dependencies
+
+RUN apk add --no-cache --virtual .build-deps make gcc g++ python \
```
Comment (AvivRubys) [inline services/registry/Dockerfile]: What is this for?

Comment (AleF83) [reply services/registry/Dockerfile]: node-gyp

File: services/registry/builtin-resources/schemas/base.gql [hunk 6]
```diff
@@ -24,7 +24,7 @@ directive @select(path: [String!]!) on FIELD_DEFINITION
 directive @gql(
     url: String!
     queryName: String!
-    arguments: String
+    arguments: String = ""
```
Comment (AvivRubys) [inline services/registry/builtin-resources/schemas/base.gql]: Why is this a string? Shouldn't it be some kind of key-value mapping?

File: e2e/tests/11_graphql_directive_tests.spec.ts [hunk 7]
```diff
@@ -48,22 +49,60 @@ describe("Graphql Directive tests", () => {
...
+                        reviews: reviews.filter(r => r.hotelId === o.hotelId).map(r => ({
+                            author: r.author,
+                            text: r.text,
+                        })),
+                    },
+                })),
+            },
         };
 
         const response = await client.request(`{
```
Comment (AvivRubys) [inline e2e/tests/11_graphql_directive_tests.spec.ts]: Use `graphql-tag` so you can use variables easily without escaping your own strings and such

File: services/registry/builtin-resources/schemas/base.gql (hunk 6 above)
Comment (AleF83) [reply services/registry/builtin-resources/schemas/base.gql]: I'll take care of this in next PR.

File: services/graphql-server/utils/ast2sdl.go (hunk 4 above)
Comment (AleF83) [reply services/graphql-server/utils/ast2sdl.go]: I'll take care of this in next PR.

File: e2e/tests/11_graphql_directive_tests.spec.ts (hunk 7 above)
Comment (AleF83) [reply e2e/tests/11_graphql_directive_tests.spec.ts]: I'll take care of it in next PR. I want to add support for non-graphql resolvers under graphql directive anyway.

**Pull request merged**